src/com/github/toxuin/griswold/EventListener.java

```diff
@@ -45,55 +46,51 @@ public void onPlayerInteractEntity(PlayerInteractEntityEvent event) {
             || !event.getPlayer().hasPermission("griswold.enchant")) {
             return;
         }
-
-        for (Repairer rep : npcChunks.keySet()) {
-            if (!event.getRightClicked().equals(rep.getEntity())) continue;
-            plugin.interactor.interact(event.getPlayer(), rep);
-            event.setCancelled(true);
-        }
+        npcChunks.entrySet().stream()
+                .filter(e -> e.getKey().getEntity().equals(event.getRightClicked()))
+                .map(Map.Entry::getKey).collect(Collectors.toList())
+                .forEach(npc -> { // TODO: -> findAny? is that possible?
```

Comment:
Are there any non unique possible matches?

Reply:
Anyway there's just one new interaction is possible per event 🤔

Reply:
My thoughts exactly. Should be able to find single match and interact with it. But honestly, I think it should just work ok as is here.